### ISSUE TITLE
Having two Head components inside _document.tsx causes weird error

### ISSUE THREAD

**Issue by awareness481** [vercel/next.js#40903]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: x64
      Version: #49~20.04.1-Ubuntu SMP Thu Aug 4 19:15:44 UTC 2022
    Binaries:
      Node: 16.14.2
      npm: 8.5.0
      Yarn: 1.22.11
      pnpm: 7.9.0
    Relevant packages:
      next: 12.3.2-canary.7
      eslint-config-next: 12.3.1
      react: 18.2.0
      react-dom: 18.2.0

 **I don't see .8 yet on npm**
warn  - Latest canary version not detected, detected: "12.3.2-canary.7", newest: "12.3.2-canary.8".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue

### Describe the Bug

When accessing a page, you get a blank page and this error. 

```js
 DOMException: Failed to execute 'measure' on 'Performance': The mark 'beforeRender' does not exist.
    at en (http://localhost:3000/_next/static/chunks/main-facca8e96989083b.js:1:17631)
    at http://localhost:3000/_next/static/chunks/main-facca8e96989083b.js:1:18490
    at Array.forEach (<anonymous>)
    at http://localhost:3000/_next/static/chunks/main-facca8e96989083b.js:1:18463
    at u2 (http://localhost:3000/_next/static/chunks/framework-03665ce53a58c61e.js:9:92095)
    at oa (http://localhost:3000/_next/static/chunks/framework-03665ce53a58c61e.js:9:98841)
    at e (http://localhost:3000/_next/static/chunks/framework-03665ce53a58c61e.js:9:111709)
    at http://localhost:3000/_next/static/chunks/framework-03665ce53a58c61e.js:9:111720
    at o8 (http://localhost:3000/_next/static/chunks/framework-03665ce53a58c61e.js:9:112231)
    at oj (http://localhost:3000/_next/static/chunks/framework-03665ce53a58c61e.js:9:103020)
um @ framework-03665ce53a58c61e.js:9
```

Related: https://github.com/vercel/next.js/issues/20743

### Expected Behavior

no error

### Link to reproduction

https://github.com/awareness481/next-two-head-components

### To Reproduce

1. Run build command
2. Run start command
3. Visit index page
4. Observe error

( or run `pnpm create next-app` and copy-paste [_document.tsx](https://github.com/awareness481/next-two-head-components/blob/main/pages/_document.tsx) )

**Comment by rubytree33** [comment]
I've looked at this a bit and it is still an issue on `12.3.2-canary.11`.

Apparently, the rendering process for `Head` is necessary for the rest of the page to function. For each `Head` component that is added to the custom `Document`, another render of the page is done. These don't necessarily happen in order, and since they all use the same performance marks, one `Head` can reset these marks before another `Head` is done with them.

It seems like the current implementation of `Head` doesn't account for being used multiple times at all, and I'm not sure it's supposed to be, since it's not the same as `Head` from `next/head`. [Relevant documentation](https://nextjs.org/docs/advanced-features/custom-document) doesn't say it shouldn't be used this way, but I assume it shouldn't.

I'll try adding a more descriptive error and/or ESLint rule for when multiple `Head`s are used in a custom `Document`. The docs might also need an update.

**Comment by ryan-williams** [comment]
I believe I'm seeing this with one `next/document` Head in `pages/_document.js` and one `next/head` Head in my `pages/index.tsx`, basically the same as in @awareness481's repro repo [awareness481/next-two-head-components](https://github.com/awareness481/next-two-head-components), except without duplicating the `<Head>` in `_document`.

My `_document.js` is just:
```javascript
import {Head, Html, Main, NextScript} from 'next/document'

export default function Document() {
    return (
        <Html>
            <Head>
                <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
            </Head>
            <body>
            <Main />
            <NextScript />
            </body>
        </Html>
    )
}
```

I tried moving that `<link>` into the `<Head>` in `pages/index.tsx` but that causes it to not take effect on the page. I can try to isolate a repro, I suspect it started when I moved from Next 12 to 13 recently.

**Comment by huozhi** [comment]
The fix is now landed in stable version 13.5.1
